### PR TITLE
idxd-initcontainer: Drop libkmod, libudev

### PR DESCRIPTION
**Pull request by ozhuraki**
- Make libkmod, libudev optional
- Include accel-config, libjson-c, libuuid sources

### CODE REVIEW

File: build/docker/intel-idxd-initcontainer.Dockerfile [hunk 1]
```diff
@@ -12,38 +12,46 @@
...
-RUN apt-get update && apt-get install -y \
-            build-essential autoconf automake autotools-dev libtool \
-            pkgconf asciidoc xmlto uuid-dev libjson-c-dev libkmod-dev \
-            libudev-dev libkeyutils-dev curl
+RUN echo "deb-src http://deb.debian.org/debian unstable main" >> \
+        /etc/apt/sources.list.d/deb-src.list && \
+    apt update && apt install -y --no-install-recommends \
+        gcc make patch autoconf automake libtool pkg-config \
+        libjson-c-dev uuid-dev curl ca-certificates && \
+    cd /usr/local/src/ && apt --download-only source uuid libjson-c5 && cd /
```
Comment (mythi) [inline build/docker/intel-idxd-initcontainer.Dockerfile:22]: Go packages are copied under `/usr/local/share/package-sources`. Let's use that here too

File: build/docker/intel-idxd-initcontainer.Dockerfile [hunk 2]
```diff
@@ -12,38 +12,46 @@
...
 
-RUN curl -fsSL "$ACCEL_CONFIG_DOWNLOAD_URL" -o accel-config.tar.gz \
-    && echo "$ACCEL_CONFIG_SHA256 accel-config.tar.gz" | sha256sum -c - \
-    && tar -xzf accel-config.tar.gz \
-    && rm accel-config.tar.gz
+RUN curl -fsSL "$ACCEL_CONFIG_DOWNLOAD_URL" -o accel-config.tar.gz && \
+    echo "$ACCEL_CONFIG_SHA256 accel-config.tar.gz" | sha256sum -c - && \
+    tar -xzf accel-config.tar.gz
+
+ADD demo/accel-config-kmod.patch /
```
Comment (mythi) [inline build/docker/intel-idxd-initcontainer.Dockerfile:32]: time to add `demo/idxd-config` (or `accel-config`) sub-directory for all intel-idxd-initcontainer related content?

**Review by bart0sh — COMMENTED** [review]
/lgtm

**Comment by ozhuraki** [comment]
@mythi 
Corrected, please take a look:
- Sources are copied to /usr/local/share/package-sources/
- Moved all intel-idxd-initcontainer related content into demo/idxd-config/

**Comment by mythi** [comment]
> Moved all intel-idxd-initcontainer related content into demo/idxd-config/

Looks like we need the Dockerfile there too and update `ci.yaml` accordingly

**Comment by ozhuraki** [comment]
Changes:
- Renamed idxd-initcontainer to idxd-config-initcontainer as discussed
- Undone move of idxd-init.sh, dsa.conf to demo/idxd-config/ (to avoid the duplication of Dockerfile for the CI to pass)

**Review by mythi — APPROVED** [review]

**Review by bart0sh — APPROVED** [review]

**Pull request merged**